Bottom: 4b48d0ce43be23955352db3c5c731937dc81fc15
Top:    ef9208fbb535507d11d5f9f2cadfbf5adf4e3276
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-14 15:33:30 -0500

Report build status to Gitlab


---
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index d9b8f54b..72733708 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -20,7 +20,8 @@ RUN pip --no-cache-dir install \
     'buildbot-waterfall-view' \
     'buildbot-www' \
     'ldap3' \
-    'pyOpenSSL'
+    'pyOpenSSL' \
+    'txrequests'
 
 # Create buildbot user
 ARG BUILDBOT_UID=1000
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index 1ac46a1a..32b0b5c0 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -60,6 +60,15 @@ mn = reporters.MailNotifier(
     )
 c['services'].append(mn)
 
+gl = reporters.GitLabStatusPush(
+    os.getenv('GITLAB_TOKEN'),
+    context="Buildbot CI",
+    baseURL=os.getenv("GITLAB_URI"),
+    verbose=True,
+    debug=True
+)
+c['services'].append(gl)
+
 ####### WEB SERVER
 userInfoProvider = util.LdapUserInfo(
         uri=os.getenv('LDAP_URI'),
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index ef79a5b9..b5134a96 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -51,6 +51,7 @@ container_env:
   GITLAB_URI: 'https://{{ gitlab_hostname }}/'
   GITLAB_CLIENT_ID: '{{ gitlab_buildbot_auth_id }}'
   GITLAB_CLIENT_SECRET: '{{ gitlab_buildbot_auth_secret }}'
+  GITLAB_TOKEN: '{{ gitlab_buildbot_token }}'
   GITLAB_WEBHOOK_SECRET: '{{ buildbot_webhook_secret }}'
   LDAP_URI: 'ldap://{{ openldap_hostname }}:389/'
   LDAP_BIND_USER: 'cn=proxyagent,dc=novatech'
